cw_nano/lib/nano_client.dart

```diff
@@ -0,0 +1,454 @@
+import 'dart:async';
+import 'dart:convert';
+
+import 'package:cw_nano/nano_balance.dart';
+import 'package:cw_nano/nano_transaction_model.dart';
+import 'package:cw_nano/nano_util.dart';
+import 'package:http/http.dart' as http;
+import 'package:nanodart/nanodart.dart';
+import 'package:cw_core/node.dart';
+
+class NanoClient {
+  static const String DEFAULT_REPRESENTATIVE =
+      "nano_38713x95zyjsqzx6nm1dsom1jmm668owkeb9913ax6nfgj15az3nu8xkx579";
+
+  Node? _node;
+  Node? _powNode;
+
+  bool connect(Node node) {
+    try {
+      _node = node;
+      return true;
+    } catch (e) {
+      return false;
+    }
+  }
+
+  bool connectPow(Node node) {
+    try {
+      _powNode = node;
+      return true;
+    } catch (e) {
+      return false;
+    }
+  }
+
+  Future<NanoBalance> getBalance(String address) async {
+    final response = await http.post(
+      _node!.uri,
+      headers: {"Content-Type": "application/json"},
+      body: jsonEncode(
+        {
+          "action": "account_balance",
+          "account": address,
+        },
+      ),
+    );
+    final data = await jsonDecode(response.body);
+    final String currentBalance = data["balance"] as String;
+    final String receivableBalance = data["receivable"] as String;
+    final BigInt cur = BigInt.parse(currentBalance);
+    final BigInt rec = BigInt.parse(receivableBalance);
+    return NanoBalance(currentBalance: cur, receivableBalance: rec);
+  }
+
+  Future<dynamic> getAccountInfo(String address) async {
+    try {
+      final response = await http.post(
+        _node!.uri,
+        headers: {"Content-Type": "application/json"},
+        body: jsonEncode(
+          {
+            "action": "account_info",
+            "representative": "true",
+            "account": address,
+          },
+        ),
+      );
+      final data = await jsonDecode(response.body);
+      return data;
+    } catch (e) {
+      print("error while getting account info");
+      rethrow;
+    }
+  }
+
+  Future<String> changeRep({
+    required String privateKey,
+    required String repAddress,
+    required String ourAddress,
+  }) async {
+    try {
+      final accountInfo = await getAccountInfo(ourAddress);
+
+      // construct the change block:
+      Map<String, String> changeBlock = {
+        "type": "state",
+        "account": ourAddress,
+        "previous": accountInfo["frontier"] as String,
+        "representative": repAddress,
+        "balance": accountInfo["balance"] as String,
+        "link": "0000000000000000000000000000000000000000000000000000000000000000",
+        "link_as_account": "nano_1111111111111111111111111111111111111111111111111111hifc8npp",
```

Comment:
link_as_account is just the link in account form
link here refers to the previous block in the block lattice, which normally refers to a receive block's corresponding send block
in this case it's null (represented as all 0's) because it's a change block and there is no corresponding send/receive